test/RootCloseWrapperSpec.js

```diff
@@ -100,6 +106,53 @@ describe('RootCloseWrapper', function () {
 
       expect(outerSpy).to.have.not.been.called;
       expect(innerSpy).to.have.been.calledOnce;
+
+      assert.oneOf(innerSpy.getCall(0).args[0].type, ['click', 'mousedown']  );
+    });
+  }
+
+  function shouldCloseOnKeyUp(eventName) {
+    it('should close when escape keyup', () => {
+      let spy = sinon.spy();
+      render(
+        <RootCloseWrapper onRootClose={spy}>
+          <div id='my-div'>hello there</div>
+        </RootCloseWrapper>
+      );
+
+      expect(spy).to.not.have.been.called;
+
+      simulant.fire(document.body, eventName, {keyCode: 27});
+
+      expect(spy).to.have.been.calledOnce;
+
+      assert.equal(spy.getCall(0).args[0].type, 'keyup' );
+      assert.equal(spy.getCall(0).args[0].keyCode, 27 );
     });
+
+  // TODO: Update code to make this pass
+  //   it('should close when inside another RootCloseWrapper', () => {
+  //         let outerSpy = sinon.spy();
+  //         let innerSpy = sinon.spy();
+
+  //         render(
+  //           <RootCloseWrapper onRootClose={outerSpy}>
+  //             <div>
+  //               <div id='my-div'>hello there</div>
+  //               <RootCloseWrapper onRootClose={innerSpy}>
+  //                 <div id='my-other-div'>hello there</div>
+  //               </RootCloseWrapper>
+  //             </div>
+  //           </RootCloseWrapper>
+  //         );
+
+  //         simulant.fire(document.body, eventName, {keyCode: 27});
+
+  //         expect(outerSpy).to.have.not.been.called;
+  //         expect(innerSpy).to.have.been.calledOnce;
```

Comment:
Agreed it should be fixed since this test exposes and an existing issue.

I'm afraid I don't have the time to commit to it. Since it is an existing bug I would say either open an issue where it can be addressed or someone willing to champion the scope change of this PR should take it on.